zbus/src/connection/mod.rs

```diff
@@ -974,20 +974,24 @@ impl Connection {
                     }) {
                         if let Some(conn) = weak_conn.upgrade() {
                             let hdr = msg.header();
-                            match hdr.destination() {
-                                // Unique name is already checked by the match rule.
-                                Some(BusName::Unique(_)) | None => (),
-                                Some(BusName::WellKnown(dest)) => {
-                                    let names = conn.inner.registered_names.lock().await;
-                                    // destination doesn't matter if no name has been registered
-                                    // (probably means name it's registered through external means).
-                                    if !names.is_empty() && !names.contains_key(dest) {
-                                        trace!(
-                                            "Got a method call for a different destination: {}",
-                                            dest
-                                        );
-
-                                        continue;
+                            // If we're connected to a bus, skip the destination check as the
+                            // server will only send method calls for us.
```

Comment:
Just a nitpick on the language:

```suggestion
                            // server will only send us method calls that are destined to us.
```